src/context.cpp

```diff
@@ -378,38 +412,93 @@ namespace librealsense
         return std::make_shared<platform_camera>(ctx, _uvcs, this->get_device_data(), register_device_notifications);
     }
 
+#ifdef BUILD_WITH_DDS
+    // This is the rs2 device; it proxies to an actual DDS device that does all the actual
+    // work. For example:
+    //     auto dev_list = ctx.query_devices();
+    //     auto dev1 = dev_list[0];
+    //     auto dev2 = dev_list[0];
+    // dev1 and dev2 are two different rs2 devices, but they both go to the same DDS source!
+    //
+    class dds_device_proxy : public software_device
+    {
+        std::shared_ptr< dds::dds_device > _dds_dev;
+
+    public:
+        dds_device_proxy( std::shared_ptr< context > ctx, std::shared_ptr< dds::dds_device > const & dev )
```

Comment:
Why not increasing the shared_ptr ref count?

Reply:
Not sure I follow

Reply:
Discussed. 👍